File: 495. Teemo Attacking/495.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        cur_time = 0
        total = 0
        for i in range(len(timeSeries)):
            if i == 0:
                cur_time = timeSeries[i]
            else:
                if timeSeries[i] > cur_time+duration-1:
                    total += duration
                elif timeSeries[i] <= cur_time+duration-1:
                    total += timeSeries[i]-cur_time
                cur_time = timeSeries[i]
        total += duration
        return total

File: 495. Teemo Attacking/495.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int curTime = 0, total = 0;
        for (size_t i=0; i<timeSeries.size(); i++) {
            if (i == 0) {
                curTime = timeSeries[i];
            }
            else {
                if (timeSeries[i] > curTime+duration-1) {
                    total += duration;
                }
                else {
                    total += (timeSeries[i]-curTime);
                }
                curTime = timeSeries[i];
            }
        }
        total += duration;
        return total;
    }
};